packages/cozy-realtime/src/RetryManager.js

```diff
@@ -0,0 +1,142 @@
+import MicroEE from 'microee'
+
+import {
+  maxWaitBetweenRetries,
+  baseWaitAfterFirstFailure,
+  timeBeforeSuccessfull,
+  raiseErrorAfterAttempts
+} from './config'
+
+class RetryManager {
+  constructor() {
+    this.reset()
+    this.onSuccess = this.onSuccess.bind(this)
+    this.onFailure = this.onFailure.bind(this)
+    this.onSuccessAcknowledged = this.onSuccessAcknowledged.bind(this)
+  }
+
+  /**
+   * On a success
+   *
+   * If `timeBeforeSuccessfull` if not null, it will wait this amount of
+   * time without errors before acknowleding a success
+   */
+  onSuccess() {
+    if (timeBeforeSuccessfull > 0) {
+      this.startSuccessWatcher(
+        this.onSuccessAcknowledged,
+        timeBeforeSuccessfull
+      )
+    } else {
+      this.onSuccessAcknowledged()
+    }
+  }
+
+  /**
+   * Acknowledge a success
+   *
+   * @private
+   */
+  onSuccessAcknowledged() {
+    this.emit('success')
+    this.reset()
+  }
+
+  /**
+   * Clears the success watcher
+   *
+   * @private
+   */
+  clearSuccessWatcher() {
+    if (this.successWatcher) {
+      global.clearTimeout(this.successWatcher)
+      this.successWatcher = null
+    }
+  }
+
+  /**
+   * Starts the success watcher - acknowledges the success if not interrupted after some time
+   *
+   * @private
+   * @param {function} ack - will be called after the wait time
+   * @param {number} time - waiting time
+   */
+  startSuccessWatcher(ack, time) {
+    if (!this.successWatcher) {
+      const callback = () => {
+        this.successWatcher = null
+        ack()
+      }
+      this.successWatcher = global.setTimeout(callback, time)
```

Comment:
Why not calling setTimeout directly (without global. ) ?

Reply:
I used to always be explicit when getting a function or attribute from the global context.